.github/workflows/cypress.yml

```
@@ -0,0 +1,37 @@
+name: E2E test
+
+on:
+  push:
+    branches:
+      - develop
+      - trunk
+  pull_request:
+    branches:
+      - develop
+jobs:
+  cypress:
+    name: ${{ matrix.core.name }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        core:
+          - {name: 'WP latest', version: 'latest'}
+          - {name: 'WP minimum', version: 'WordPress/WordPress#5.7'}
+          - {name: 'WP trunk', version: 'WordPress/WordPress#master'}
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
```

It would be good to use a later version, I suggest 4.2.2 to match the version used in other actions in this repository.

In https://github.com/10up/block-catalog/pull/79 we switched to pinning action workflows to particular hashes rather than using tags or version numbers. This was following the [compromise of `tj-actions/changed-files`](https://semgrep.dev/blog/2025/popular-github-action-tj-actionschanged-files-is-compromised/).

```suggestion
      uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
```